Using CSS counters - add reversing examples

hamishwillee opened the pull request:
This updates [using CSS counters](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Counter_Styles/Using_CSS_counters) (and `counter-reset` a bit) with `reversed()` counter examples.

hamishwillee committed:
Using CSS counters - add reversing examples

sideshowbarker commented on files/en-us/web/css/counter-reset/index.md:
```diff
@@ -16,14 +16,20 @@ This property will create a new counter or reversed counter with the given name
+Using `counter-reset` with the `list-item` counter that controls numbering on ordered lists is slightly different.
+Firstly, the default initial `list-item` counter value is zero for forward counters and the number of elements in the list for reversed counters.
+Secondly, by default the `list-item` counter automatically increments or decrements by one for each element.
+This makes it very easy to when working with ordered lists to count up from zero or (reversed) down from the number of elements to one.
 
```
```suggestion
This makes it very easy when working with ordered lists to count up from zero or (reversed) down from the number of elements to one.

```

hamishwillee committed:
Using CSS counters/counter reset - clarify defaults

hamishwillee commented:
@sideshowbarker If you feel inclined to review this again, note that it has significant update since last time.

The new version makes it clear that the main thing about reversed counters is that they have a default initial value reflecting the number of elements for which numbering is to be applied. That makes it easy to implement a counter that goes backwards from the number of elements to 1. But you still have to do that decrementing yourself unless you're using a `list-item` counter.

estelle commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
```diff
@@ -12,21 +12,64 @@ tags:
+The counter is rendered in the specified style (`decimal` by default).
+
+The {{cssxref("counters()")}} function also has two forms: 'counters(_name_, _string_)' or 'counters(_name_, _string_, _style_)'.
+The generated text is the value of all counters with the given name in scope at the given pseudo-element, from outermost to innermost, separated by the specified string. The counters are rendered in the specified style (`decimal` by default).
```
Can this be made more understandable? Perhaps with an example?

hamishwillee replied:
Thanks @estelle . You're right. Despite the fact this is shown in green it was inherited from before I got to this section and I didn't look at it properly. Next week. 

Thanks for the review!

sideshowbarker replied:
Has this resolved be subsequent updates?

hamishwillee replied:
Yes.

estelle committed:
changed order so example comes after explanation

moved the example which includes using a default value to be after the default value is explained.

estelle commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
```diff
@@ -12,21 +12,66 @@ tags:
+Like other counters this has a default initial value of 0 for upward counters and "number of items" for reversed counters.
+Unlike author-created counters, `list-item` _automatically_ increments or decrements by one for each list element, depending on whether or not the counter is reversed.
+
+This counter can be used to manipulate the default behavior of ordered lists using CSS.
```
which counter?

hamishwillee replied:
I thought it was inferred since this is the only counter we're talking about in this section.

```suggestion
The `list-item` counter can be used to manipulate the default behavior of ordered lists using CSS.
```

estelle commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
```diff
@@ -59,9 +104,46 @@ h3::before {
+This example is the same as the one above but uses a reversed counter.
+If your browser supports the `reversed()` function notation, the result will look like this:
+
+![reversed counter](reversed_headings_basic.png)
```
alt attribute

hamishwillee replied:
@estelle Markdown doesn't support an alt attribute right?

sideshowbarker replied:
` ![reversed counter](reversed_headings_basic.png)` does cause an `alt="reversed counter"` attribute in the DOM, right?

hamishwillee replied:
I think so, but you don't do anything special extra to get it. In any case, if it is a problem it is one that is not specific to this PR.

hamishwillee replied:
I'm marking this resolved.

estelle commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
```diff
@@ -59,9 +104,46 @@ h3::before {
 ### A more sophisticated example
 
-A counter must not necessarily be shown every time is incremented. This example counts all links, however the counter is shown only when a link has no text, as a convenient replacement.
+A counter need not necessarily be shown every time is incremented.
```
```suggestion
A counter need not necessarily be shown every time it is incremented.
```

estelle commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
```diff
@@ -59,9 +104,46 @@ h3::before {
 
-A counter must not necessarily be shown every time is incremented. This example counts all links, however the counter is shown only when a link has no text, as a convenient replacement.
+A counter need not necessarily be shown every time is incremented.
+This example counts all links, however the counter is shown only when a link has no text, as a convenient replacement.
```
```suggestion
This example counts all links with the counter showing only when a link has no text, as a convenient replacement.
```

hamishwillee committed:
Update files/en-us/web/css/css_counter_styles/using_css_counters/index.md

hamishwillee committed:
Apply suggestions from code review

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

hamishwillee committed:
Better manipulating counter examples

hamishwillee committed:
Slightly better structure

hamishwillee commented:
@estelle @sideshowbarker Thanks for the review. I updated in line with suggestions. Note that in the section on displaying counters I mostly concentrated on a single simple example of one case. At the end I link to the respective examples at the end.

sideshowbarker commented on files/en-us/web/css/counter-reset/index.md:
```diff
@@ -90,6 +85,10 @@ The "implicit" counter named `list-item` can be used to control the numbering fo
 
 ## Examples
 
+The following examples show how reset the counters, but not how they are incremented, decremented, and displayed.
```
```suggestion
The following examples show how to reset the counters, but not how they are incremented, decremented, and displayed.
```

sideshowbarker commented on files/en-us/web/css/counter-reset/index.md:
```diff
@@ -90,6 +85,10 @@ The "implicit" counter named `list-item` can be used to control the numbering fo
 
 ## Examples
 
+The following examples show how reset the counters, but not how they are incremented, decremented, and displayed.
+
+For more complete examples see [Using CSS Counters](/en-US/docs/Web/CSS/CSS_Counter_Styles/Using_CSS_counters).
+
```
```suggestion
The following examples show how to reset the counters, but not how they are incremented, decremented, and displayed.

For more-complete examples see [Using CSS Counters](/en-US/docs/Web/CSS/CSS_Counter_Styles/Using_CSS_counters).

```

sideshowbarker commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
```diff
@@ -12,21 +12,130 @@ tags:
+To use a CSS counter, it must first be initialized to a value with the {{cssxref("counter-reset")}} property.
+The property can also be used to change the counter value to any specific number.
+
+Below we initialise a counter named `section` to the default value (0).
```
```suggestion
Below we initialize a counter named `section` to the default value (0).
```

sideshowbarker replied:
```suggestion
Below we initialize a counter named `section` to the default value (0).
```

sideshowbarker commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
````diff
@@ -12,21 +12,130 @@ tags:
+To use a CSS counter, it must first be initialized to a value with the {{cssxref("counter-reset")}} property.
+The property can also be used to change the counter value to any specific number.
+
+Below we initialise a counter named `section` to the default value (0).
+
+```css
+counter-reset: section;
+```
+
+You can also initialise multiple counters, optionally specifying an initial value for each.
+Below we initialize the `section` and `topic` counters to the default value, and the `page` counter to 3.
+
+```css
+counter-reset: section page 3 topic;
+```
+
+Once initialized, a counter's value can be increased or decreased using {{cssxref("counter-increment")}}.
+For example, the following declaration would increment the `section` counter by one on every `h3` tag.
+
+```css
+h3::before {
+  counter-increment: section; /* Increment the value of section counter by 1 */
+}
+```
+You can specify the value to increment or decrement the counter after the counter name, using a positive or negative number.
+
+The counter's name must not be `none`, `inherit`, or `initial`; otherwise the declaration is ignored.
````
```suggestion
Below we initialize a counter named `section` to the default value (0).

```css
counter-reset: section;
```

You can also initialize multiple counters, optionally specifying an initial value for each.
Below we initialize the `section` and `topic` counters to the default value, and the `page` counter to 3.

```css
counter-reset: section page 3 topic;
```

Once initialized, a counter's value can be increased or decreased using {{cssxref("counter-increment")}}.
For example, the following declaration would increment the `section` counter by one on every `h3` tag.

```css
h3::before {
  counter-increment: section; /* Increment the value of section counter by 1 */
}
```
You can specify the value to increment or decrement the counter after the counter name, using a positive or negative number.

The counter's name must not be `none`, `inherit`, or `initial`; otherwise the declaration is ignored.
```

sideshowbarker commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
````diff
@@ -12,21 +12,130 @@ tags:
+counter-reset: section;
+```
+
+You can also initialise multiple counters, optionally specifying an initial value for each.
````
```suggestion
You can also initialize multiple counters, optionally specifying an initial value for each.
```

sideshowbarker commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
````diff
@@ -12,21 +12,130 @@ tags:
+```
+
+The {{cssxref("counters()", "counters()")}} function is used when the count for nested levels must include the count from parent levels.
+For example, you might use this to layout sections as shown:
````
```suggestion
For example, you might use this to lay out sections as shown:
```

sideshowbarker commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
```diff
@@ -12,21 +12,130 @@ tags:
+Reversed counters have a default initial value equal to the number of elements (unlike normal counters, which have a default value of 0).
+This makes it easy to implement a counter that counts from the number of elements down to one.
+
+For example, to create a reversed counter named `section` with a default initial value you would use the following syntax:
```
```suggestion
For example, to create a reversed counter named `section` with a default initial value, you would use the following syntax:
```

sideshowbarker commented on files/en-us/web/css/css_counter_styles/using_css_counters/index.md:
```diff
@@ -12,21 +12,130 @@ tags:
+
+Ordered lists, as created using {{HTMLElement("ol")}} elements, implicitly have a counter named `list-item`.
+
+Like other counters this has a default initial value of 0 for upward counters and "number of items" for reversed counters.
```
```suggestion
Like other counters, this has a default initial value of 0 for upward counters and "number of items" for reversed counters.
```

sideshowbarker committed:
Apply suggestions from code review

hamishwillee commented:
Thanks for the reviews. @estelle I'm fairly confident your issues have been addressed or were invalid. So I'm going to merge. However happy to take comments against this PR and do another one if I am wrong.
